app/lib/actions/katello/repository/sync.rb

```diff
@@ -56,7 +56,7 @@ def plan(repo, options = {})
               concurrence do
                 plan_action(Katello::Repository::MetadataGenerate, repo, :force => true) if skip_metadata_check && repo.yum?
                 plan_action(Katello::Repository::ErrataMail, repo, nil, contents_changed)
-                plan_action(Actions::Katello::Applicability::Repository::Regenerate, :repo_ids => [repo.id]) if generate_applicability
+                plan_action(Actions::Katello::Applicability::Repository::Regenerate, :repo_ids => [repo.id]) if generate_applicability || repo.deb?
```

Comment:
generate_applicability is defined above to fall back on `repo.yum?` if the param isn't given to the action. that could be updated to `repo.yum? || repo.deb?` to consolidate such checking - removing the one you added here